# Don't do comment in comment. It doesn't work...
# The original code fails on AMD CPUs with AVX+FMA4
# Patch matches the code in Python 3.6.4
#
# Lars Viklund
diff -ru Python-3.6.3.orig/Modules/_blake2/impl/blake2s-load-xop.h Python-3.6.3/Modules/_blake2/impl/blake2s-load-xop.h
--- Python-3.6.3.orig/Modules/_blake2/impl/blake2s-load-xop.h	2017-10-03 07:52:02.000000000 +0200
+++ Python-3.6.3/Modules/_blake2/impl/blake2s-load-xop.h	2018-02-05 16:39:28.221091877 +0100
@@ -18,8 +18,9 @@
 
 #define TOB(x) ((x)*4*0x01010101 + 0x03020100) /* ..or not TOB */
 
+#if 0
 /* Basic VPPERM emulation, for testing purposes */
-/*static __m128i _mm_perm_epi8(const __m128i src1, const __m128i src2, const __m128i sel)
+static __m128i _mm_perm_epi8(const __m128i src1, const __m128i src2, const __m128i sel)
 {
    const __m128i sixteen = _mm_set1_epi8(16);
    const __m128i t0 = _mm_shuffle_epi8(src1, sel);
@@ -27,7 +28,8 @@
    const __m128i mask = _mm_or_si128(_mm_cmpeq_epi8(sel, sixteen),
                                      _mm_cmpgt_epi8(sel, sixteen)); /* (>=16) = 0xff : 00 */
    return _mm_blendv_epi8(t0, s1, mask);
-}*/
+}
+#endif
 
 #define LOAD_MSG_0_1(buf) \
 buf = _mm_perm_epi8(m0, m1, _mm_set_epi32(TOB(6),TOB(4),TOB(2),TOB(0)) );
